plutus-core/plutus-ir/src/PlutusIR/Transform/Inline/Inline.hs

```diff
@@ -212,17 +212,21 @@ processTerm = handleTerm <=< traverseOf termSubtypes applyTypeSubstitution where
             -- actually have got rid of all of them!
             pure $ mkLet ann NonRec bs' t'
         -- This includes recursive let terms, we don't even consider inlining them at the moment
-        t ->
-            -- process all subterms first, so that the rhs won't be processed more than once. This
-            -- is important because otherwise the number of times we process them can grow
-            -- exponentially in the case that it has nested `let`s.
-            --
-            -- Then, consider call site inlining for each node that have gone through unconditional
-            -- inlining. Because `inlineSaturatedApp` traverses *all* application nodes for each
-            -- subterm, the runtime is quadratic for terms with a long chain of applications.
-            -- If we use the context-based approach like in GHC, this won't be a problem, so we may
-            -- consider that in the future.
-            inlineSaturatedApp =<< forMOf termSubterms t processTerm
+        t -> do
+            {- We consider call site inlining when we see an application node, then process the term
```

Comment:
This is not accurate - we don't do callsite inlining before unconditional inlining (and we shouldn't). Unconditional inlining is still being done first - if a binding should be unconditionally inlined, then it won't get to the callsite inliner.

Reply:
Yeah, I think this should just stick to saying "process the subterms" and drop the bit saying it's unconditional inlining, which it isn't. It's running the whole combined inliner on the subterms.